### PR TITLE
fix: adjust number formatters + "Max" spending limit

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Number formatting and "Max" spending limit.

## How this PR fixes it

The number formatters have been updated to use `ethers`' `formatUnits` instead of `toDecimals` and the "Max" token amount has been removed from the `<SpendingLimitForm>` field as it's possible to set higher amounts than that which is in the Safe.

## How to test it

1. Observe all numbers formatted correctly across the Safe, i.e. in the assets list, in high and low amount transactions and in modals. Choosing the "Max" amount when the current transaction has high or low amounts, formatted with `>` or `<` should not copy these symbols into their respective amount fields.
2. The "Max" button should no longer exist in the spending limits form.

### CODE REVIEW

File: src/utils/formatters.ts [hunk 1]
```diff
@@ -2,12 +2,8 @@ import { BigNumberish, type BigNumber } from 'ethers'
 import { formatUnits, parseUnits } from 'ethers/lib/utils'
 import { formatAmount } from './formatNumber'
 
-export const toDecimals = (value: BigNumberish, decimals?: number | string): number => {
-  return Number(formatUnits(value, decimals))
-}
-
```
Comment (iamacook) [inline src/utils/formatters.ts:8]: This was an unnnecessary wrapper as we only once need convert the formatted numbers to numbers.

File: src/components/common/SafeTokenWidget/index.tsx [hunk 2]
```diff
@@ -23,9 +23,8 @@ const SafeTokenWidget = () => {
 
   const safeBalance = balances.balances.items.find((balanceItem) => balanceItem.tokenInfo.address === tokenAddress)
 
-  const safeBalanceDecimals = Math.floor(
-    toDecimals(safeBalance?.balance || '0', safeBalance?.tokenInfo.decimals),
-  ).toFixed(0)
+  const safeBalanceDecimals = Number(formatUnits(safeBalance?.balance || 0, safeBalance?.tokenInfo.decimals))
+  const flooeredSafeBalance = Math.floor(safeBalanceDecimals).toFixed(0)
```
Comment (katspaugh) [inline src/components/common/SafeTokenWidget/index.tsx:27]: Typo in floored

**Comment by katspaugh** [comment]
A tests is failing.

What are the differences between formatDecimals and formatUnits?

**Comment by iamacook** [comment]
> What are the differences between formatDecimals and formatUnits?

@usame-algan and I just went through every instance of number formatting and cleaned everything up a bit to be the following:

- `safeFormatUnits` for anything logic-related, returning `123`, `0.00000000001`, etc.
- `safeFormatAmount` for anything UI-related, returning `-< 0.00001`, `999T`, etc.
- `safeParseUnits` for converting to `BigNumbers`.

The following formatters were removed as the above now covers their logic:
- `toDecimals` = `safeFormatUnits`
- `formatDecimals` = `safeFormatAmount`
- `toWei` = `safeParseUnits`

**Comment by katspaugh** [comment]
Awesome! To make it more obvious which function to use – safeFormatUnits vs safeFormatAmount – should we maybe rename `safeFormatAmount` to `visualFormatAmount` or `renderAmount` or some such?

**Comment by iamacook** [comment]
> Awesome! To make it more obvious which function to use – safeFormatUnits vs safeFormatAmount – should we maybe rename `safeFormatAmount` to `visualFormatAmount` or `renderAmount` or some such?

Great suggestion! I've renamed it to `formatVisualAmount` and adjusted the explanatory comment above.

**Review by katspaugh — APPROVED** [review]

File: src/utils/formatters.ts [hunk 3]
```diff
@@ -2,29 +2,27 @@ import { BigNumberish, type BigNumber } from 'ethers'
...
-  return formatAmount(toDecimals(value, decimals))
-}
-
-export const toWei = (value: string, decimals?: number | string): BigNumber => {
-  return parseUnits(value, decimals)
-}
-
 const GWEI = 'gwei'
 
+// safeFormatUnits => "0.000000000001"
```
Comment (schmanu) [inline src/utils/formatters.ts:7]: I would suggest to write a proper comment here which explains the difference between `formatVisualAmount` and `safeFormatUnits`.

i.e.
```ts
/**
 *  Formats a GWEI value to decimal with high precision. 
 *  The resulting string is not formatted with respect to the standard amount formatting. Use {@link formatVisualAmount} to format values with amount formatting.
 * @param value in GWEI
 * @param decimals decimals of the value in GWEI, 18 decimals are used if undefined
 * @returns formatted decimal value with high precision
 */
```

File: src/utils/formatters.ts [hunk 4]
```diff
@@ -2,29 +2,27 @@ import { BigNumberish, type BigNumber } from 'ethers'
...
+    // FIXME: Temporary fix to as ethers' `formatFixed` doesn't strip trailing 0s
+    // https://github.com/5afe/safe/wiki/How-to-format-amounts
+    return parseFloat(formattedAmount).toString()
   } catch (err) {
     console.error('Error formatting units', err)
     return ''
   }
 }
 
+// formatVisualAmount => -< 0.00001
```
Comment (schmanu) [inline src/utils/formatters.ts:21]: Here I would prefer a similar comment.
These functions are easily mistaken.

**Review by schmanu — COMMENTED** [review]
It's good that we streamlined the amount /value formatting! Nice work!

**Pull request merged**